Add StartupTimeout to xService resource

mgr32 opened the pull request:
Currently it is hardcoded to wait only 2 s for a service to start. I believe this can be often not enough - this commit adds 'StartupTimeout' parameter (in milliseconds) with default 30 s.

KarolKaczmarek commented on DSCResources/MSFT_xServiceResource/MSFT_xServiceResource.psm1:
```diff
@@ -322,12 +327,12 @@ function Set-TargetResource
          if(( $PSBoundParameters.ContainsKey("State")) -and ($State -eq "Running"))
         {
-            StartService $svc
+            StartService $svc $StartupTimeout
```
Please don't use positional parameters. Specify parameter names and their values explicitly

KarolKaczmarek commented on DSCResources/MSFT_xServiceResource/MSFT_xServiceResource.psm1:
```diff
@@ -322,12 +327,12 @@ function Set-TargetResource
     else
     {
-        StartService $svc
+        StartService $svc $StartupTimeout
```
Please don't use positional parameters. Specify parameter names and their values explicitly

KarolKaczmarek commented:
Looks good, I left minor comments about positional parameters. We want to adhere to PSScriptAnalyzer rules and use of positional parameters is discouraged by it.

mgr32 commented:
I certainly agree but didn't want to break consistency as positional parameters were used everywhere.  In second commit I fixed it in whole file, and also fixed parameter names casing.

KarolKaczmarek commented:
Awesome!
